### PR TITLE
Auto docs

### PR DESCRIPTION
**Pull request by jreidinger**
see http://yast-pkg-bindings.surge.sh for current version

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,4 +3,5 @@
 [![Travis Build](https://travis-ci.org/yast/yast-pkg-bindings.svg?branch=master)](https://travis-ci.org/yast/yast-pkg-bindings)
 [![Jenkins Build](http://img.shields.io/jenkins/s/https/ci.opensuse.org/yast-pkg-bindings-master.svg)](https://ci.opensuse.org/view/Yast/job/yast-pkg-bindings-master/)
 
+Online documentation is available at [yast-pkg-bindings.surge.sh/](http://yast-pkg-bindings.surge.sh/)
```
Comment (lslezak) [inline README.md]: I'd prefer the HTTPS variant here...

File: Doxyfile [hunk 2]
```diff
@@ -0,0 +1,2477 @@
+# Doxyfile 1.8.12
```
Comment (lslezak) [inline Doxyfile]: Is this the original generated file or was something manually changed (besides the obvious `PROJECT_NAME`)?

The manual changes should be documented so we do not loose them after regenerating the file.

File: Rakefile [hunk 3]
```diff
@@ -4,3 +4,10 @@ Yast::Tasks.configuration do |conf|
   #lets ignore license check for now
   conf.skip_license_check << /.*/
 end
+
+task :doc do
+  rm_rf "autodocs"
+  mkdir "autodocs"
```
Comment (lslezak) [inline Rakefile]: Check the `git status`, I guess `autodocs` is missing in `.gitignore`...

File: doc/generate_index.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+# @return [Array] list of builtns
+def builtins
+  files = Dir.glob(File.expand_path("../../src/*.{h,cc}", __FILE__))
+  res = []
+  files.each do |source_path|
+    lines = File.readlines(source_path)
+    puts "processing #{source_path}"
+    lines.each do |l|
+      case l
+      when /@builtin/
```
Comment (lslezak) [inline doc/generate_index.rb]: Um, `case`/`when` with only a single branch? Why not simple `if l =~ /@builtin/` ?

File: Doxyfile (hunk 2 above)
Comment (jreidinger) [reply Doxyfile]: originally generated, but manual changes in INPUT and macros. File should not be regenerated later. it is just updated when doxygen is newer, but it is old school software, so not allows regeneration.

File: README.md (hunk 1 above)
Comment (jreidinger) [reply README.md]: ok

File: Rakefile (hunk 3 above)
Comment (jreidinger) [reply Rakefile]: will add it

File: doc/generate_index.rb (hunk 4 above)
Comment (jreidinger) [reply doc/generate_index.rb]: my plan is to add in future also detection of params and maybe also return to have there correct type. This is just first shot to have something, that can be improved in future.

Comment (lslezak) [reply doc/generate_index.rb]: OK, then please at least add `TODO` marks so it's obvious why it's there.

File: README.md (hunk 1 above)
Comment (lslezak) [reply README.md]: JFYI: do not trust anything transferred via HTTP (unless you can verify it somehow), see e.g. https://bugzilla.suse.com/show_bug.cgi?id=1052272

File: Doxyfile (hunk 2 above)
Comment (lslezak) [reply Doxyfile]: If we do not touch it then OK.

**Pull request merged**